test/e2e/memcached_test.go

```diff
@@ -82,25 +83,42 @@ func TestMemcached(t *testing.T) {
 		t.Fatalf("Error: %v\nCommand Output: %s\n", err, string(cmdOut))
 	}
 
+	// get global framework variables
+	f := framework.Global
 	t.Log("Building operator docker image")
-	cmdOut, err = exec.Command("operator-sdk",
-		"build",
-		"quay.io/example/memcached-operator:v0.0.1").CombinedOutput()
-	if err != nil {
-		t.Fatalf("Error: %v\nCommand Output: %s\n", err, string(cmdOut))
-	}
-	operatorYAML, err := ioutil.ReadFile("deploy/operator.yaml")
-	if err != nil {
-		t.Fatal(err)
-	}
-	operatorYAML = bytes.Replace(operatorYAML, []byte("imagePullPolicy: Always"), []byte("imagePullPolicy: Never"), 1)
-	err = ioutil.WriteFile("deploy/operator.yaml", operatorYAML, os.FileMode(filemode))
-	if err != nil {
-		t.Fatal(err)
+	if *f.ImageName == "" {
+		cmdOut, err = exec.Command("operator-sdk",
+			"build",
+			"quay.io/example/memcached-operator:v0.0.1").CombinedOutput()
+		if err != nil {
+			t.Fatalf("Error: %v\nCommand Output: %s\n", err, string(cmdOut))
+		}
+		operatorYAML, err := ioutil.ReadFile("deploy/operator.yaml")
+		if err != nil {
+			t.Fatal(err)
+		}
+		operatorYAML = bytes.Replace(operatorYAML, []byte("imagePullPolicy: Always"), []byte("imagePullPolicy: Never"), 1)
+		err = ioutil.WriteFile("deploy/operator.yaml", operatorYAML, os.FileMode(filemode))
+		if err != nil {
+			t.Fatal(err)
+		}
+	} else {
```

Comment:
Why does the else block have different logic than that of the if block beside building a different named image?
e.g:
* The if block sets up `deploy/operator.yaml` but else block doesn't.
* The else block push the docker image but if block doesn't.

Reply:
If the user doesn't specify an image, we assume that they are building an image to a local repo in a local minikube instance (you can share the internal repo with `eval $(minikube docker-env)`. This is the case with the travis builds as well. In this case, we need to set the `imagePullPolicy` to `Never`. Otherwise, the operator initialization would stall trying to pull the latest image from a remote repo.

If a user specifies an image name, we assume that the user has push access to the remote repository specified, so we push the image to the remote repo and keep the `imagePullPolicy` set to the default `Always` to make sure that their kubernetes instance gets the latest image for the test.

Other than setting the image name in a new variable and then using that to remove duplication of the `build` command, the logic needs to remain separate unless we want to add another command line option specifying if the user wants to push the image remotely or use the local docker repo instead.

Reply:
I see. that makes. Would the following be more concise and more readable?

```go
	local := *f.ImageName == ""
	if local {
		*f.ImageName = "quay.io/example/memcached-operator:v0.0.1"
	}
	cmdOut, err = exec.Command(
		"operator-sdk",
		"build",
		*f.ImageName).CombinedOutput()
	if err != nil {
		t.Fatalf("Error: %v\nCommand Output: %s\n", err, string(cmdOut))
	}
	operatorYAML, err := ioutil.ReadFile("deploy/operator.yaml")
	if err != nil {
		t.Fatal(err)
	}
	if local {
		operatorYAML = bytes.Replace(operatorYAML, []byte("imagePullPolicy: Always"), []byte("imagePullPolicy: Never"), 1)
		t.Log("Pushing docker image to repo")
		cmdOut, err = exec.Command("docker",
			"push",
			*f.ImageName).CombinedOutput()
		if err != nil {
			t.Fatalf("Error: %v\nCommand Output: %s\n", err, string(cmdOut))
		}
	}
	err = ioutil.WriteFile("deploy/operator.yaml", operatorYAML, os.FileMode(filemode))
	if err != nil {
		t.Fatal(err)
	}
```

Reply:
Yes, similar to that (changing the operatorYAML and pushing the docker image need to be separate though). I'm pushing a change for this now.

Reply:
yeah, you are right. I missed the part about ` the docker image need to be separate though)`